### PR TITLE
Design System Alert

### LINKED ISSUE #214: Design system - Alerts

### PR DESCRIPTION
**Pull request by hanihusam**
Closes #214 

## Description

Create reusable Alert component

## Current Tasks
- [x] Create the base component
- [x] Allow to add icon
- [x] Make it dissmisable
- [x] Testing for the component

### CODE REVIEW

**Comment by hanihusam** [comment]
Mas @resir014 @zainfathoni I think we need global types for colors type and props. It is required by badge and alert. Wdyt?

**Comment by resir014** [comment]
@hanihusam Yes, let's go with that.

**Comment by resir014** [comment]
Can we also replace the alert in the homepage with this new alert, so we can see what it's like? @hanihusam 

image

**Comment by hanihusam** [comment]
> Can we also replace the alert in the homepage with this new alert, so we can see what it's like? @hanihusam 
> 
> 
> 
> image
> 
> 

I think I already change it with the new one. Please kindly check it.

File: components/ui/alert/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+) {
+  if (icon) {
+    return React.createElement(icon, {
+      className: clsx("h-5 w-5", additionalClasses),
+      "aria-hidden": true,
+    });
+  }
+
+  return null;
+}
```
Comment (zainfathoni) [inline components/ui/alert/utils/helpers.ts:87]: As you can see in the warning messages from Codecov above, you need to test this file either directly (by writing a unit test for this file) or indirectly (by writing an integration test for the home page where this file is eventually used). Please write the tests for your new components. You can look at some examples in the existing `*.test.ts` or `*.test.tsx` files. 😁 
Thanks!

File: pages/index.tsx [hunk 2]
```diff
@@ -36,38 +36,24 @@ const htmrTransform: HtmrOptions["transform"] = {
...
-            minute: "2-digit",
-            timeZone: "Asia/Jakarta",
-            timeZoneName: "short",
-          })}
-        </p>
-      </div>
-    </div>
-  </div>
+const LastUpdatedAlert = ({ className }: LastUpdatedAlertProps) => (
+  <Alert accentBorder className={className} icon={ClockIcon}>
```
Comment (zainfathoni) [inline pages/index.tsx:42]: This line is also needs to be covered by your tests.

**Review by zainfathoni — CHANGES_REQUESTED** [review]

File: pages/index.tsx (hunk 2 above)
Comment (zainfathoni) [reply pages/index.tsx:42]: I have added a test for the `HomePage`, but apparently it's not sufficient to test all the possibilities in the `helpers.ts` file. 😅

File: components/ui/alert/alert.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      ...rest
+    },
+    ref,
+  ) => {
+    const [_visible, setVisible] = React.useState(visible);
+
+    const _className = clsx(
+      alertColors(color),
+      accentBorderColors(accentBorder, color),
+      "rounded-md p-4",
```
Comment (resir014) [inline components/ui/alert/alert.tsx:35]:
```suggestion
      "p-4",
```

I think we can live without `border-radius` from now on. The upcoming designs for Web 2.0 will be without rounded corners too, so let's follow the same design.

File: components/ui/alert/utils/types.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+   * Method called before the dissmiss animation has started.
+   */
+  onDismiss?: () => void;
+  /**
+   * Method called after the dissmiss animation has completed and the component is removed from the dom.
+   */
+  onDismissed?: () => void;
+  visible?: boolean;
+  /**
+   * Set broder style in the left of component.`
```
Comment (resir014) [inline components/ui/alert/utils/types.ts:34]:
```suggestion
   * Set border style in the left of component.`
```

typo 😅

File: components/ui/types.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export type Colors =
```
Comment (resir014) [inline components/ui/types.ts:1]:
```suggestion
export type BaseColors =
```

Be more specific with the type naming.

File: package.json [hunk 6]
```diff
@@ -42,6 +42,7 @@
     "preact": "^10.5.14",
     "react": "17.0.2",
     "react-dom": "17.0.2",
+    "react-transition-group": "^4.4.2",
```
Comment (resir014) [inline package.json:45]:
```suggestion
    "react-transition-group": "^4.4.2",
```

Use `Transition` in [Headless UI](https://headlessui.dev/react/transition) instead.

**Review by resir014 — DISMISSED** [review]

File: components/ui/alert/utils/helpers.ts (hunk 1 above)
Comment (zainfathoni) [reply components/ui/alert/utils/helpers.ts:87]: I have added a unit test for the `helpers.ts` file.
Please update the tests accordingly when you're updating your implementations based on @resir014's comments.
Thanks!

**Comment by resir014** [comment]
@hanihusam also, when #487 is merged, we will have an additional colour, `light-blue`. When the time comes, please update this PR accordingly.

**Review by zainfathoni — APPROVED** [review]

**Pull request merged**